sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/FileIndexSuite.scala

```diff
@@ -520,6 +520,21 @@ class FileIndexSuite extends SharedSparkSession {
       SQLConf.get.setConf(StaticSQLConf.METADATA_CACHE_TTL_SECONDS, previousValue)
     }
   }
+
+  test("SPARK-34314: preserve partition values of the string type") {
```

Comment:
This doesn't seem like a UT of `FileIndex`. Shall we move it to `SQLQuerySuite`?

Reply:
Maybe, `datasources/parquet/ParquetPartitionDiscoverySuite.scala` since it has end-2-end tests already? `SQLQuerySuite` is big enough and gathers everything.

Reply:
SGTM